packages/documentation-v7/src/stories/components/switch/switch.stories.ts

```diff
@@ -0,0 +1,237 @@
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'General',
+      },
+    },
+    checked: {
+      name: 'Checked',
+      description: 'When set to `true`, places the component in the checked state.',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'States',
+      },
+    },
+    disabled: {
+      name: 'Disabled',
+      description:
+        'When set to `true`, disables the component\'s functionality and places it in a disabled state.<span className="mt-mini alert alert-info alert-sm">There are accessibility concerns with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessibility--page#disabled-state">disabled state accessibility guide</a>.</span>',
+      control: {
+        type: 'boolean',
+      },
+      table: {
+        category: 'States',
+      },
+    },
+    validation: {
+      name: 'Validation',
+      description: "Controls the display of the component's validation state.",
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid',
+        },
+      },
+      options: ['null', 'is-valid', 'is-invalid'],
+      table: {
+        category: 'States',
+      },
+    },
+  },
+};
+
+export default meta;
+
+const VALIDATION_STATE_MAP: Record<string, undefined | boolean> = {
+  'null': undefined,
+  'is-valid': false,
+  'is-invalid': true,
+};
+
+function getValidationFeedback({ validation }: Args) {
+  return validation !== 'null'
+    ? html`
+        <p class=${validation.split('-')[1] + '-feedback'}>
+          ${validation === 'is-valid' ? 'Ggranda sukceso!' : 'Eraro okazis!'}
+        </p>
+      `
+    : null;
+}
+
+function renderSwitch(args: Args, context: StoryContext) {
+  const [_, updateArgs] = useArgs();
+
+  const switchClasses = mapClasses({
+    'form-check-input': true,
+    [args.validation]: args.validation !== 'null',
+  });
+
+  const useAriaLabel = args.hiddenLabel;
+  const useLabelBefore = !useAriaLabel && ['before', 'both'].includes(args.labelPosition);
+  const useLabelAfter = !useAriaLabel && ['after', 'both'].includes(args.labelPosition);
+
+  const labelBefore = useLabelBefore
+    ? html`
+        <label for="switch" class="form-check-label order-first">${args.labelBefore}</label>
+      `
+    : null;
+
+  const labelAfter = useLabelAfter
+    ? html`
+        <label for="switch" class="form-check-label">${args.labelAfter}</label>
+      `
+    : null;
+
+  const handleChange = (e: Event) => {
+    updateArgs({ checked: !args.checked });
+  };
+
+  return html`
+    <div class="form-check form-switch">
+      <input
+        id=${context.id}
+        class=${switchClasses}
+        type="checkbox"
+        role="switch"
+        ?checked=${args.checked}
```

Comment:
This should help, to not break the control, if a user is checking/unchecking the switch in the story.
```suggestion
        ?checked=${args.checked}
        .checked=${args.checked}
```